src/mono/wasm/wasm.proj

```diff
@@ -252,4 +240,38 @@
           SkipUnchangedFiles="true" />
   </Target>
 
+  <Target Name="InstallRollup"
+          Inputs="$(MonoProjectRoot)wasm/runtime/package.json"
+          Outputs="$(MonoProjectRoot)wasm/runtime/node_modules/.package-lock.json"
+          >
+    <!-- install typescript and rollup -->
+    <RunWithEmSdkEnv Condition="'$(ContinuousIntegrationBuild)' == 'true'" Command="npm ci" EmSdkPath="$(EMSDK_PATH)" IgnoreStandardErrorWarningFormat="true" WorkingDirectory="$(MonoProjectRoot)wasm/runtime/"/>
+    <RunWithEmSdkEnv Condition="'$(ContinuousIntegrationBuild)' == 'true'" Command="npm audit" EmSdkPath="$(EMSDK_PATH)" IgnoreStandardErrorWarningFormat="true" WorkingDirectory="$(MonoProjectRoot)wasm/runtime/"/>
+    <!-- npm install is faster on dev machine as it doesn't wipe node_modules folder -->
+    <RunWithEmSdkEnv Condition="'$(ContinuousIntegrationBuild)' != 'true'" Command="npm install" EmSdkPath="$(EMSDK_PATH)" IgnoreStandardErrorWarningFormat="true" WorkingDirectory="$(MonoProjectRoot)wasm/runtime/"/>
+
+    <Touch Files="$(MonoProjectRoot)wasm/runtime/node_modules/.package-lock.json" />
+  </Target>
+
+
+  <ItemGroup>
+    <_RollupInputs Include="$(MonoProjectRoot)wasm/runtime/*.ts"/>
+    <_RollupInputs Include="$(MonoProjectRoot)wasm/runtime/types/*.ts"/>
+    <_RollupInputs Include="$(MonoProjectRoot)wasm/runtimetypes/*.d.ts"/>
+    <_RollupInputs Include="$(MonoProjectRoot)wasm/runtime/*.json"/>
+    <_RollupInputs Include="$(MonoProjectRoot)wasm/runtime/*.js"/>
+  </ItemGroup>
+
+  <Target Name="BuildWithRollup"
+          Inputs="@(_RollupInputs)"
+          Outputs="$(NativeBinDir)dotnet.d.ts.sha256"
+          >
+    <!-- code style check -->
+    <RunWithEmSdkEnv Command="npm run lint" StandardOutputImportance="High" EmSdkPath="$(EMSDK_PATH)" WorkingDirectory="$(MonoProjectRoot)wasm/runtime/"/>
+
+    <!-- compile typescript -->
+    <RunWithEmSdkEnv Command="npm run rollup -- --environment Configuration:$(Configuration),NativeBinDir:$(NativeBinDir),ProductVersion:$(ProductVersion)" EmSdkPath="$(EMSDK_PATH)" IgnoreStandardErrorWarningFormat="true" WorkingDirectory="$(MonoProjectRoot)wasm/runtime/"/>
+
+    <Touch Files="$(NativeBinDir)dotnet.d.ts.sha256" />
```

Comment:
Do we need the touch here? Doesn't the command above create this file?

Reply:
It does, but it doesn't touch it when it's same as before. 
I could also make it to touch it every time instead, but this seemed more obvious to me.